utility/servicer/module_test.go

```diff
@@ -43,63 +75,151 @@ func TestAdmitRelay(t *testing.T) {
 		},
 		{
 			name:     "Relay for unsupported chain is rejected",
-			relay:    testRelay("foo", 8),
+			relay:    testRelay(testRelayChain("foo")),
 			expected: errValidateRelayMeta,
 		},
 		{
 			name:     "Relay with height set in a past session is rejected",
-			relay:    testRelay("0021", 5),
+			relay:    testRelay(testRelayHeight(5)),
 			expected: errValidateBlockHeight,
 		},
 		{
 			name:     "Relay with height set in a future session is rejected",
-			relay:    testRelay("0021", 9999),
+			relay:    testRelay(testRelayHeight(9999)),
 			expected: errValidateBlockHeight,
 		},
+		{
+			name:     "Relay not matching the servicer in this session is rejected",
+			relay:    testRelay(testRelayServicer("bar")),
+			expected: errValidateServicer,
+		},
+		{
+			name:              "Relay for app out of quota is rejected",
+			relay:             testRelay(),
+			usedSessionTokens: 999999,
+			expected:          errValidateApplication,
+		},
 	}
 
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
 			config := testServicerConfig()
 			session := testSession(
-				sessionNumber(2),
+				sessionNumber(currentSessionNumber),
 				sessionBlocks(4),
-				sessionHeight(2),
+				sessionHeight(testSessionStartingHeight),
 				sessionServicers(testServicer1),
 			)
-			mockBus := mockBus(t, &config, currentHeight, session)
+			mockBus := mockBus(t, &config, uint64(testCurrentHeight), session, testCase.usedSessionTokens)
 
 			servicerMod, err := CreateServicer(mockBus)
 			require.NoError(t, err)
 
-			err = servicerMod.(*servicer).admitRelay(testCase.relay)
+			servicer, ok := servicerMod.(*servicer)
+			require.True(t, ok)
+
+			err = servicer.admitRelay(testCase.relay)
 			if !errors.Is(err, testCase.expected) {
 				t.Fatalf("Expected error %v got: %v", testCase.expected, err)
 			}
 		})
 	}
 }
 
-func testRelay(chain string, height int64) *coreTypes.Relay {
-	return &coreTypes.Relay{
+func TestExecuteRelay(t *testing.T) {
+	testCases := []struct {
+		name        string
+		relay       *coreTypes.Relay
+		expectedErr error
+	}{
+		{
+			name:        "relay is rejected if chain is not specified in the config",
+			relay:       testRelay(testRelayChain("foo")),
+			expectedErr: errValidateRelayMeta,
+		},
+		{
+			name:  "Relay for accepted chain is executed",
+			relay: testRelay(),
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				fmt.Fprintln(w, `{"0x1234"}`)
+			}))
+			defer ts.Close()
+
+			config := testServicerConfig()
+			for ch := range config.Services {
+				config.Services[ch].Url = ts.URL
+			}
+
+			servicer := &servicer{config: &config}
+			_, err := servicer.executeRelay(testCase.relay)
+			if !errors.Is(err, testCase.expectedErr) {
+				t.Fatalf("Expected error %v got: %v", testCase.expectedErr, err)
+			}
```

Comment:
Why not use: https://pkg.go.dev/github.com/stretchr/testify/assert#ErrorIs

Reply:
Thank you for catching this. Fixed.